### ISSUE TITLE
Guidance on how to set a `party/id` when the source data doesn't provide one

### ISSUE THREAD

**Issue by yolile** [open-contracting/standard#1635]
This question has been raised by at least 3 partners this year (India, Canada and Guatemala), so we need to add some guidance on the best alternatives to fill this field (and what to do with `Organization/identifier`)

**Comment by jpmckinney** [comment]
We can perhaps have two streams:

* Input: Constrain user input, i.e. come up with an authoritative list of buyers, and ideally assign them identifiers so that if a name is changed (which occurs frequently in Canada, for example) the identifier remains the same (Canada actually has a Federal Identity Program, so that the civil service can use one consistent name internally, independent of whatever name the current governing party prefers)

* Mapping: If input can't be constrained, apply normalizations so that small differences in names don't result in new identifiers, e.g. lowercase, normalize spaces, remove articles, prepositions and conjunctions (I've observed lots of variations around these), substitute common abbreviations and typos (min. -> ministry, etc.). Recommend that they produce a full list of names, so that they can identify other things to normalize. It won't be perfect, but better than no normalization.

When I worked with Quebec data many years ago, one ministry had something like 34 variations.

**Comment by yolile** [comment]
Agree, but note that the first one will only work for buyers but not for suppliers.
For example, in Guatemala, there are some cases where they don't have the identifier for foreign companies, and only their names are recorded.

**Comment by jpmckinney** [comment]
Ah, yeah, since there are different options for each case, I suppose the guidance could be organized by type of organization:

1. buyer or procuring entity
2. registered supplier
3. unregistered supplier

And then within each:

1. Control user input (codelist for buyers, company registry or other supplier database for registered suppliers)
2. Normalize available data

For suppliers, normalization is trickier, because small differences in names can actually be two different companies. In some cases, the best available option might just be to assign a local ID, and not attempt to make it a global ID.

**Comment by yolile** [comment]
Yeap, that is what I thought, too. I marked this issue as Documentation, but I'm not sure if this should be a Worked example under the "Deal with the hard cases" section instead
